kevm-pyk/src/kevm_pyk/utils.py

```diff
@@ -91,9 +93,90 @@ def get_apr_proof_for_spec(
     return apr_proof
 
 
+class APRProofStrategy(ABC):
+    _execute_depth: int
+    _terminal_rules: Iterable[str]
+    _cut_point_rules: Iterable[str]
+    _counterexample_info: bool
+    _always_check_subsumption: bool
+    _fast_check_subsumption: bool
```

Comment:
This I assume is shared between both strategies, Consider extracting a class for passing them around in one object. That should get rid of `*args, **kwargs` as well.

Reply:
Added `RunProverParams` dataclass for this.